modules/openapi-generator/src/main/resources/Javascript/es6/ApiClient.mustache

```diff
@@ -404,6 +409,25 @@ class ApiClient {
 
         request.query(this.normalizeParams(queryParams));
 
+    {{#detectAmznApiGatewayExtensions}}
+        // Retrieve host from baseURL and full path from baseURL + path
+        var host = this.basePath.substring(8);
+        var fullPath = host.substring(host.indexOf('/') + 1);
+        host = host.substring( 0, host.indexOf( fullPath ) - 1);
+        fullPath += path;
+
+        // Set credentials to null if user is not providing
+        var credentials = null;
+        if (accessKey !== undefined && secretKey !== undefined){
```

Comment:
NIT: also in javascript you can do this

```
if (accessKey && secretKey)
```

This would basically check for null, undefined and empty all cases

Reply:
JFYI

https://stackoverflow.com/questions/5515310/is-there-a-standard-function-to-check-for-null-undefined-or-blank-variables-in

Reply:
Excellent, made the change.